lib/beaker/hypervisor/abs.rb

```diff
@@ -74,5 +77,65 @@ def provision
     def cleanup
       # nothing to do
     end
+
+    def provision_vms(hosts)
+      ns_request, vm_request = generate_floaty_request_strings(hosts)
+      # TODO Implement ns_request
+
+      vm_beaker_abs = []
+      # HACKY NEED TO PROPERLY ACCESS VMFLOATY
+      # Will return a JSON Object like this:
+      # {"redhat-7-x86_64"=>["rich-apparition.delivery.puppetlabs.net", "despondent-side.delivery.puppetlabs.net"], "centos-7-x86_64"=>["firmer-vamp.delivery.puppetlabs.net"]}
+      vm_floaty_output = JSON.parse(`floaty --service=abs get #{vm_request} --json`)
```

Comment:
Calling floaty like this is just an example, need to clean this up I think before properly turning this into a PR.

Reply:
looking into that. Most likely will need a small change in the vmfloaty class.

Reply:
As a side note, instead of the change proposed, you can already hook into the floaty class by using it as a gem.

```ruby
require 'vmfloaty/abs'
```

But I also get it if you'd rather have the CLI class build up the right parameters from command line arguments and pass them along. The downside here is you are going to get vmloaty output in your job, and you might be subject to breaking changes if the CLI output ever changes. If you use the class directly, you should hopefully not deal with those. But there's some additional work up front you'll need to do that the command line class does.